scripts/lib/CIME/XML/component.py

```diff
@@ -116,14 +118,74 @@ def _get_value_match(self, node, attributes=None, exact_match=False):
         return match_value
 
     #pylint: disable=arguments-differ
-    def get_description(self, compsetname):
+    def get_description(self, compsetname, comp_class=None):
```

Comment:
This method now serves the additional important purpose of doing error checking on a compset name, right? If so, I'd vote for renaming it, even if the name becomes more verbose: something like get_compset_description_and_check_validity